lib/charms/vault_k8s/v0/vault_tls.py

```diff
@@ -0,0 +1,451 @@
+        Returns:
+            str: path
+        Raises:
+            VaultCertsError: If the CA certificate is not found
+        """
+        storage = self.charm.model.storages
+        if "certs" not in storage:
+            raise VaultCertsError()
+        if not storage["certs"]:
+            raise VaultCertsError()
+        cert_storage = storage["certs"][0]
+        storage_location = cert_storage.location
+        return f"{storage_location}/{file.name.lower()}.pem"
+
+    def _get_ca_certificate_secret_in_peer_relation(self) -> Tuple[str, str]:
+        """Get the vault CA certificate secret from the peer relation.
+
+        Returns:
+            Tuple[Optional[str], Optional[str]]: The CA private key and certificate
+        """
+        try:
+            peer_relation = self.charm.model.get_relation(self.peer_relation)
+            juju_secret_id = peer_relation.data[peer_relation.app].get(
+                CA_CERTIFICATE_JUJU_SECRET_KEY
+            )
+            juju_secret = self.charm.model.get_secret(id=juju_secret_id)
+            content = juju_secret.get_content()
+            return content["privatekey"], content["certificate"]
+        except (TypeError, SecretNotFoundError, AttributeError):
+            raise PeerSecretError(secret_name=CA_CERTIFICATE_JUJU_SECRET_KEY)
+
+    def _set_ca_certificate_secret_in_peer_relation(
+        self,
+        private_key: str,
+        certificate: str,
+    ) -> None:
+        """Set the vault CA certificate secret in the peer relation.
+
+        Args:
+            private_key: Private key
+            certificate: certificate
+        """
+        if not self.charm._is_peer_relation_created():
+            raise RuntimeError("Peer relation not created")
+        juju_secret_content = {
+            "privatekey": private_key,
+            "certificate": certificate,
+        }
+        juju_secret = self.charm.app.add_secret(
+            juju_secret_content, label=CA_CERTIFICATE_JUJU_SECRET_LABEL
+        )
+        peer_relation = self.charm.model.get_relation(self.peer_relation)
+        peer_relation.data[self.charm.app].update({CA_CERTIFICATE_JUJU_SECRET_KEY: juju_secret.id})
+        logger.debug("Vault CA certificate secret set in peer relation")
+
+    def _remove_ca_certificate_secret_in_peer_relation(self) -> None:
+        """Removes the vault CA certificate secret in the peer relation.
+
+        Only the key from the relation is removed. The actual juju secret is unmodified.
+        """
+        if not self.charm._is_peer_relation_created():
+            raise RuntimeError("Peer relation not created")
+        peer_relation = self.model.get_relation(self.peer_relation)
+        if peer_relation:
+            peer_relation.data[self.charm.app].pop(CA_CERTIFICATE_JUJU_SECRET_KEY)
+            logger.debug("Vault CA certificate secret removed from peer relation")
+
+    def _ca_certificate_set_in_peer_relation(self) -> bool:
```

Comment:
This is a public method now as it is called by the charm code, make sure to remove the `_` prefix in these cases

Reply:
Updated.